src/lib/address/ens.ts

```diff
@@ -0,0 +1,68 @@
+// @ts-ignore TODO: add type declarations for ensjs module
+import ENS, { getEnsAddress } from '@ensdomains/ensjs'
+import { ChainAdapterManager } from '@shapeshiftoss/chain-adapters'
+import { ChainTypes } from '@shapeshiftoss/types'
+import { getConfig } from 'config'
+import memoize from 'lodash/memoize'
+import { ethChainId } from 'test/mocks/accounts'
+import { getChainAdapters } from 'context/PluginProvider/PluginProvider'
+import { getWeb3Provider } from 'lib/web3-provider'
+
+import { ReverseLookupVanityDomain } from './address'
+import { ResolveVanityDomain, ResolveVanityDomainReturn, ValidateVanityDomain } from './address'
+
+let makeEns: () => void
+// getEnsAddress takes a magic number as string, networkId. 1 stands for mainnet
+const ens = new Promise<void>(resolve => (makeEns = resolve)).then(async () => {
+  const unchainedUrls = {
+    ethereum: {
+      httpUrl: getConfig().REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL,
+      wsUrl: getConfig().REACT_APP_UNCHAINED_ETHEREUM_WS_URL,
+    },
+  }
+
+  const ethereumChainAdapter = new ChainAdapterManager(unchainedUrls).byChain(ChainTypes.Ethereum)
+  const chainId = ethereumChainAdapter.getChainId()
+  const chainIdReference = chainId.match(
+    /^(?<chainIdNamespace>[-a-z0-9]{3,8}):(?<chainIdReference>[-a-zA-Z0-9]{1,32})$/,
+  )?.groups?.chainIdReference
+  return new ENS({ provider: getWeb3Provider(), ensAddress: getEnsAddress(chainIdReference) })
+})
+
+export const resolveEnsDomain: ResolveVanityDomain = async ({ value }) =>
+  (await getChainAdapters().byChainId(ethChainId).validateAddress(value)).valid
```

Comment:
Isn't the check already happening here? https://github.com/shapeshift/web/pull/1948/files#diff-14ad33c87d46e8417b7708a7e24e20808c361c450f2dcae009d11ba3bda158c9R139
```

Regardless, this isn't just resolving an ENS domain at this point as it has an `orElse` of sorts, returning back the passed value if it's a valid address
// at this point it's a valid vanity address, let's resolve it
```

Reply:
confirmed - nothing else uses this function now, it can just be simplified to ensLookup